agent/proto/resourcesmonitor/fetcher_test.go

```diff
@@ -0,0 +1,14 @@
+package resourcesmonitor_test
+
+type resourcesFetcherMock struct {
+	fetchResourcesMonitoredMemoryFunc func() (total int64, used int64, err error)
+	fetchResourcesMonitoredVolumeFunc func(volume string) (total int64, used int64, err error)
```

Comment:
In your tests you return static values for the monitored resources, so I think it's much clearer to define a struct with the static values and then have that struct implement the interface, rather than create this extra layer of indirection.